src/figuresClasses.ts

```diff
@@ -1,19 +1,97 @@
-export interface Figure {
+type Colors = 'red' | 'green' | 'blue';
+type Shapes = 'triangle' | 'circle' | 'rectangle';
 
+export interface Figure {
+  color: Colors;
+  shape: Shapes;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shapes;
 
-}
+  getArea(): number {
+    const semiPerimeter = (this.sideA + this.sideB + this.sideC) / 2;
+
+    return +Math.sqrt(Math.abs(semiPerimeter
+      * (semiPerimeter - this.sideA)
+      * (semiPerimeter - this.sideB)
+      * (semiPerimeter - this.sideC))).toFixed(2);
+  }
+
+  isInvalidShape(): boolean {
+    const arr = [this.sideA, this.sideB, this.sideC];
+
+    const maxSide = arr.sort((a, b) => a - b).pop();
+    const twoSideSum = arr.reduce((sum, current) => sum + current);
+    const isInvalidValue = arr.some((side) => side <= 0);
+
+    if (!maxSide) {
+      return false;
+    }
 
-export class Circle {
+    return (maxSide >= twoSideSum) || isInvalidValue;
+  }
 
+  constructor(
+    public color: Colors,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    this.shape = 'triangle';
```

Comment:
Maybe, let's try to create an enum with the values.

Reply:
+